web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -36,134 +38,137 @@ export function GuestsList(props: Props) {
-                   && createModalButton('shutdown', modalsData, row) }
-                  {(state === 'paused' || state === 'running') && createModalButton('restart', modalsData, row) }
-                  {props.saltEntitled && state === 'running' && (
+            const actionsProvider =
+              (row) => {
+                if (props.foreignEntitled) {
+                  return [];
+                }
+                const state = row.stateLabel;
+                return (
+                  <div className="btn-group">
+                    {state !== 'running' && row.name !== 'Domain-0'
+                     && (
+                       <AsyncButton
+                         defaultType="btn-default btn-sm"
+                         title={t(state === 'paused' ? 'Resume' : 'Start')}
+                         icon="fa-play"
+                         action={() => onAction('start', [row.uuid], {})}
+                       />) }
+                    {state === 'running' && row.name !== 'Domain-0'
+                     && createModalButton('suspend', modalsData, row) }
+                    {state !== 'stopped' && row.name !== 'Domain-0'
+                     && createModalButton('shutdown', modalsData, row) }
+                    {(state === 'paused' || state === 'running') && createModalButton('restart', modalsData, row) }
+                    {props.saltEntitled && state === 'running' && (
+                      <LinkButton
+                        title={t('Graphical Console')}
+                        className="btn-default btn-sm"
+                        icon="fa-desktop"
+                        href={`/rhn/manager/systems/details/virtualization/guests/${props.serverId}/console/${row.uuid}`}
+                        target="_blank"
+                      />
+                    )}
                     <LinkButton
-                      title={t('Graphical Console')}
+                      title={t('Edit')}
                       className="btn-default btn-sm"
-                      icon="fa-desktop"
-                      href={`/rhn/manager/systems/details/virtualization/guests/${props.serverId}/console/${row.uuid}`}
-                      target="_blank"
+                      icon="fa-edit"
+                      href={`/rhn/manager/systems/details/virtualization/guests/${props.serverId}/edit/${row.uuid}`}
                     />
-                  )}
-                  <LinkButton
-                    title={t('Edit')}
-                    className="btn-default btn-sm"
-                    icon="fa-edit"
-                    href={`/rhn/manager/systems/details/virtualization/guests/${props.serverId}/edit/${row.uuid}`}
-                  />
-                  { props.saltEntitled && row.name !== 'Domain-0'
-                    && createModalButton('delete', modalsData, row) }
-                </div>
-              );
-            }
+                    { props.saltEntitled && row.name !== 'Domain-0'
```

Comment:
Wouldn't it be worth to save the `Domain-0` to a variable you can compare against? I see this string occurs many time. I wonder the troubles if at some point it will change to something else. Am I missing something?

Reply:
`Domain-0` is fairly unlikely to change since this is the name of the physical machine on a Xen virtual host

Reply:
and a sed call once every 5-10 years isn't such a big work